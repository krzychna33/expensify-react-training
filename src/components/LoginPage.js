import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export class LoginPage extends React.Component {
    render(){
        return (
            <div>
                <h1>Expensify app</h1>
                <h4>Please log in.</h4>
                <button onClick={this.props.startLogin}>Login</button>
            </div>
        )
    }
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);