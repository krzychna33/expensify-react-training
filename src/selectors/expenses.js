import moment from 'moment'
// Get visible expenses

export default (expenses, {text, sortBy, startDate, endDate}) => {
    return expenses.filter((expense) =>{
        const createdAtMoment = moment(expense.createdAt);

        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
        const textMatch = expense.description.toLowerCase().search(text.toLowerCase()) >= 0;

        return startDateMatch && endDateMatch && textMatch;
    }).sort((a,b) =>{
        if(sortBy === 'date'){
            if(a.createdAt < b.createdAt){
                return 1;
            } else {
                return -1;
            }
        } else if (sortBy === 'amount'){
            if(a.amount < b.amount){
                return 1;
            } else {
                return -1;
            }
        }
    })
};