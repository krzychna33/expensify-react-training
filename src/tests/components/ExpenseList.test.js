import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseList } from "../../components/ExpenseList";
import expenses from '../fixtures/expenses'

test('Should render Expense List with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesList with empty message', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
})