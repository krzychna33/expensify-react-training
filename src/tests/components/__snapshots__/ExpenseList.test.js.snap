// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expense List with expenses 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <ExpenseListItem
    amount={300}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={109500}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Credit Card"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`Should render ExpensesList with empty message 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <p>
     No Expenses 
  </p>
</div>
`;
